modo/api.py

```diff
@@ -370,3 +377,28 @@ def enrich_metadata(self):
                     self.update_element(inst_names[ele.name], ele)
                 else:
                     continue
+
+    def stream_cram(
+        self, cram_name: str, region: str = None, output: str = None
+    ):
+        """Slices and streams the requested CRAM file, both local and remote,
+        and either outputs a data stream or writes data to local file"""
+
+        # check requested CRAM exists in MODO
+        if Path(cram_path) not in self.list_files():
+            raise ValueError(f"{cram_path} not found in {self.path}.")
+
+        if self.s3_endpoint:
```

Comment:
__Nitpick__(minor): I think it would be more natural to check for `htsget_endpoint` here as we are using it downstream? In principle they should always be paired, so I don't think this would break something, but feels more coherent to me?

Reply:
@cmdoret and I had the same discussion. I agree that checking self.htsget_endpoint in the same as checking self.s3_endpoint, and it won't break anything just to switch from one to the other, but for coherency of reading the code, I think it is more natural to check if there is a self.s3_endpoint, when wanting to check if the file is remote on an s3 bucket.